Flow.Launcher.Infrastructure/Image/ImageLoader.cs

```diff
@@ -113,7 +114,25 @@ private static ImageResult LoadInternal(string path, bool loadFullImage = false)
                 {
                     return new ImageResult(ImageCache[path], ImageType.Cache);
                 }
-
+                if (path.StartsWith("http://") || path.StartsWith("https://"))
+                {
+                    // Download image from url
+                    using (WebClient client = new WebClient())
```

Comment:
Could you use HttpClient instead and try to make this method async? (It's not a trivial refactor, but I think it will be a good place to understand how async work).